src/lib/parser.ts

```diff
@@ -66,6 +66,8 @@ export class Parser {
     fieldNode: ReducedFieldNode
   ): any {
     const type = ensureNullableType(value, givenType, fieldNode);
+    if (isNone(value)) return value;
```

Comment:
good point. Originally I was thinking of allowing some custom parsing on nulls on scalars. Later I decided against it but I never went back and change the code. Nice one.